src/ShopBundle/objects/TShopBasket/TShopBasketArticleCoreList.class.php

```diff
@@ -801,4 +801,31 @@ protected function PostDeleteItemHook($oDeletedItem)
             $this->aObservers[$sObserverName]->OnBasketItemDeleteEvent($oDeletedItem);
         }
     }
+
+    /**
+     * items with the same sBasketItemKey will be merged into one item.
+     */
+    public function mergeIdenticalBasketItems(): void
+    {
+        /** @var TShopBasketArticle[] $items */
+        $items = [];
+        $mergedItems = false;
+        $this->setItemPointer(0);
+        while (false !== ($item = $this->next())) {
+            $key = $item->sBasketItemKey;
+            if (isset($items[$key])) {
```

Comment:
array_key_exists

Reply:
we know that `$items` only ever contains objects - so in my eyes we have no advantage in using array_key_exists. isset has the advantage of being shorter (easier to read) and being faster.

Reply:
And is doing more than the "array key exists" that is desired here.

Reply:
if, for whatever reason, $items[$key] was null, then we would want to overwrite that array entry. So in fact, it is doing exactly what we want here :)$items[$key]